src/queries/role.js

```diff
@@ -70,6 +72,21 @@ function queryAllByRole(
           }) === false
         : true
     })
+    .filter(element => {
+      if (name === undefined) {
+        // Don't care
+        return true
+      }
+
+      const accessibleName = computeAccessibleName(element)
+      if (typeof name === 'string') {
+        return name === accessibleName
+      }
+      if (typeof name === 'function') {
+        return name(accessibleName, element)
+      }
+      return name.test(accessibleName)
```

Comment:
Why don't we not assume it's a regex and check whether it is a regex. If it's not then we can throw a more helpful error message.

Honestly, we should probably have some code already that handles TextMatch options and *that* code could do this. Don't we already have some TextMatch code?

Reply:
That is how TextMatch is treated in other instances as well:
- https://github.com/testing-library/dom-testing-library/blob/a03f05671f849802c2d890e6de4532fb6b52f28c/src/matches.js#L11-L12
- https://github.com/testing-library/dom-testing-library/blob/a03f05671f849802c2d890e6de4532fb6b52f28c/src/matches.js#L26-L27
 
We could add a runtime TypeError though nowadays I would rather let types do the heavy lifting.

Reply:
You're right we can reuse existing logic: 

```patch
diff --git a/src/queries/role.js b/src/queries/role.js
index 3d93a48..a442bfd 100644
--- a/src/queries/role.js
+++ b/src/queries/role.js
@@ -78,14 +78,12 @@ function queryAllByRole(
         return true
       }
 
-      const accessibleName = computeAccessibleName(element)
-      if (typeof name === 'string') {
-        return name === accessibleName
-      }
-      if (typeof name === 'function') {
-        return name(accessibleName, element)
-      }
-      return name.test(accessibleName)
+      return matches(
+        computeAccessibleName(element),
+        element,
+        name,
+        text => text,
+      )
     })
```

-- a542ebcf544663bed58b13d48c6a0af0d4fa9d5f